hudi-common/src/main/java/org/apache/hudi/metrics/Metrics.java

```diff
@@ -176,4 +190,16 @@ public static boolean isInitialized(String basePath) {
     }
     return false;
   }
+
+  /**
+   * Use the same base path as the hudi table so that Metrics instance is shared.
+   */
+  private static String getBasePath(HoodieMetricsConfig metricsConfig) {
+    String basePath = metricsConfig.getBasePath();
+    if (basePath.endsWith(HoodieTableMetaClient.METADATA_TABLE_FOLDER_PATH)) {
```

Comment:
can we introduce a utility for deducing metadata table. 
btw, we should check the entire dir name matches "metadata" and not just ends with. We could possible have a table named "customer_metadata" or something of those sorts. above check could actually match for this table path.

Reply:
my bad. 
```
  public static final String METADATA_TABLE_FOLDER_PATH = METAFOLDER_NAME + Path.SEPARATOR + "metadata"; 
```

looks like we already account for what I asked for